### PR TITLE
[go_lib] feat: refactor update settings

### PR DESCRIPTION
**Pull request by juev**
## Description

Migrate to a single option for update mode with the following enum:

- Manual - deckhouse downloads releases info, but does not apply them
- AutoPatch (default) - deckhouse automatically applies patch releases, but asks for approval of minor releases
- Auto - deckhouse automatically applies all releases

## Why do we need it, and what problem does it solve?

Fixed: #8572

## What is the expected result?

- Simplifying the deckhouse update process management
- No discrepancies in configuration

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: feature
summary: deckhouse update settings. were extended.
impact_level: default
```

### CODE REVIEW

**Comment by juev** [comment]
How I tested:
1. Create a release in the Deployed status.
2. Change the update mode for mc
3. Create releases of other versions with the dry-run annotation
4. Check the release statuses

### Auto mode

Screenshot 2024-08-12 at 13 14 29
Screenshot 2024-08-12 at 13 15 13
Screenshot 2024-08-12 at 13 16 07

### AutoPatch mode

Screenshot 2024-08-12 at 14 58 43
Screenshot 2024-08-12 at 15 13 57

### Manual mode

Screenshot 2024-08-12 at 15 20 17
Screenshot 2024-08-12 at 15 21 10

File: modules/002-deckhouse/openapi/doc-ru-config-values.yaml [hunk 1]
```diff
@@ -19,6 +19,9 @@ properties:
       mode:
         description: |
           Режим обновления минорных версий Deckhouse (обновление релиза). **Не влияет на обновление patch-версий (patch-релизов).**
+          - `AutoPatch` — автоматический режим обновления. Ручной режим обновления минорной версии Deckhouse (обновление релиза, например, с `v1.26.*` на `v1.27.*`) необходимо подтверждение.
+
+            Для подтверждения обновления в соответствующем [custom resource `DeckhouseRelease`](../../cr.html#deckhouserelease) необходимо установить поле `approved` в `true`.
```
Comment (z9r5) [inline modules/002-deckhouse/openapi/doc-ru-config-values.yaml:24]:
```suggestion
          - `AutoPatch` — автоматический режим обновления для патч-версий. Ручной режим обновления для минорных версий (например, обновление релиза с `v1.62.*` на `v1.63.*`).

            Для подтверждения обновления минорной версии, в соответствующем [custom resource `DeckhouseRelease`](../../cr.html#deckhouserelease) необходимо установить поле `approved` в `true`.
```

File: modules/002-deckhouse/openapi/config-values.yaml [hunk 2]
```diff
@@ -46,16 +46,20 @@ properties:
     properties:
       mode:
         type: string
-        default: 'Auto'
+        default: 'AutoPatch'
         description: |
           Deckhouse minor version update mode (release change). **Does not affect patch version updates.**
+          - `AutoPatch` – all patch releases are applied automatically. For minor updates  manual confirmation is required.
+
+          To confirm the update, edit [custom resource 'DeckhouseRelease'](../../cr.html#deckhouserelease), which corresponds to the version by setting the `approved' field to 'true'.
```
Comment (z9r5) [inline modules/002-deckhouse/openapi/config-values.yaml:54]:
```suggestion
          - `AutoPatch` – all patch releases are applied automatically. For minor updates  manual confirmation is required.

            To confirm the update, set the `approved' field to 'true' in the corresponding [custom resource 'DeckhouseRelease'](../../cr.html#deckhouserelease).
```

**Review by z9r5 — APPROVED** [review]

**Review by yalosev — APPROVED** [review]
Great job. Especially tests!

**Pull request merged**